File: src/main/java/com/lannstark/lec04/Lec04Main.java
package com.lannstark.lec04;

public class Lec04Main {

    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney money2 = money1;
        JavaMoney money3 = new JavaMoney(1_000L);

        System.out.println(money1 == money2);
        System.out.println(money1.equals(money3));
        System.out.println(money1.plus(money2));
    }

}

File: src/main/java/com/lannstark/lec04/Lec04Main.kt
package com.lannstark.lec04

fun main() {
    val money1 = JavaMoney(1000L)
    val money2 = money1
    val money3 = JavaMoney(1000L)

    println(money1.plus(money2))

    if (money1 > money2) {
        println("Money1이 Money2보다 큽니다.")
    }

    println(money1 === money2)
    println(money1 == money3)

}